qa/workunits/cephtool/test.sh

```diff
@@ -609,6 +609,20 @@ function test_auth()
   ceph auth del client.xx
   expect_false ceph auth get client.xx
 
+  # test rotation
+  ceph auth get-or-create client.admin2 mon 'allow *'
+  ceph auth get client.admin2 >> keyring1
+  env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth get client.admin2 >> keyring2
+  diff -au keyring1 keyring2
+  # rotate itself
+  env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth rotate client.admin2 >> keyring3
+  expect_false diff -au keyring1 keyring3
+  expect_false env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth get client.admin2
+  expect_true env CEPH_KEYRING=keyring3 ceph -n client.admin2 auth get client.admin2
+  expect_true ceph auth get client.admin2 >> keyring4
+  expect_true diff -au keyring3 keyring4
+  rm keyring[1234]
```

Comment:
I would love to see a check ensuring the only thing that got changed is the secret itself; particularly that the caps are untouched.

Reply:
```
diff --git a/qa/workunits/cephtool/test.sh b/qa/workunits/cephtool/test.sh
index d024ce6912d4..266f133c6037 100755
--- a/qa/workunits/cephtool/test.sh
+++ b/qa/workunits/cephtool/test.sh
@@ -613,14 +613,20 @@ function test_auth()
   ceph auth get-or-create client.admin2 mon 'allow *'
   ceph auth get client.admin2 >> keyring1
   env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth get client.admin2 >> keyring2
-  diff -au keyring1 keyring2
+  # they are the same:
+  expect_true diff -au keyring1 keyring2
   # rotate itself
   env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth rotate client.admin2 >> keyring3
-  expect_false diff -au keyring1 keyring3
+  # only the key has changed:
+  diff -au keyring1 keyring3 | grep -E '^[-+][^-+]' | expect_false grep -v key
+  # the key in keyring1 no longer works:
   expect_false env CEPH_KEYRING=keyring1 ceph -n client.admin2 auth get client.admin2
+  # the key in keyring3 should work:
   expect_true env CEPH_KEYRING=keyring3 ceph -n client.admin2 auth get client.admin2
+  # now verify the key from `auth get` matches what rotate produced:
   expect_true ceph auth get client.admin2 >> keyring4
   expect_true diff -au keyring3 keyring4
+  expect_true ceph auth rm client.admin2
   rm keyring[1234]
 
   # (almost) interactive mode
```

Reply:
:+1: Thank you!